### PR TITLE
Add Camera Settings

### PR DESCRIPTION
**Pull request by Sushma7785**
Camera settings - Get/Set
- [x] White Balance
- [x] Color Mode
- [x] Exposure Mode
- [x] Shutter Speed
- [x] ISO Value
- [x] Resolution
- [x] Photo Format
- [x] Photo Quality
- [x] Video Format
- [x] Video Quality
- [x] Metering

### CODE REVIEW

File: Yuneec_SDK_iOS/YNCSDKCamera.h [hunk 1]
```diff
@@ -24,6 +24,37 @@ typedef void (^YNCCameraCompletion)(NSError *error);
...
+    /**
+     Enhanced mode
+     */
+    YNCENHANCED,
+    /**
+     Night mode
+     */
+    YNCNIGHT,
+    /**
+     Unprocesses mode
```
Comment (julianoes) [inline Yuneec_SDK_iOS/YNCSDKCamera.h]: We should document what `Unprocessed` means. It means that no additional color processing has been performed and no artificial sharpening.

File: Yuneec_SDK_iOS/YNCSDKCamera.h [hunk 2]
```diff
@@ -24,6 +24,37 @@ typedef void (^YNCCameraCompletion)(NSError *error);
...
+ */
+typedef NS_ENUM (NSInteger, YNCCameraColorMode) {
+    /**
+     Neutral mode
+     */
+    YNCNEUTRAL = 0,
+    /**
+     Enhanced mode
+     */
+    YNCENHANCED,
```
Comment (julianoes) [inline Yuneec_SDK_iOS/YNCSDKCamera.h]: I wonder if `YNCENHANCED` is readable enough won't lead to conflicts.

I think I vote for:

```
typedef NS_ENUM (NSInteger, YNCCameraColorMode) {
   /**
    Neutral mode
    */
   YNCCameraColorModeNeutral = 0,
    ...
```

File: Yuneec_SDK_iOS/YNCSDKCamera.mm [hunk 3]
```diff
@@ -26,6 +26,41 @@ void receive_camera_result(YNCCameraCompletion completion, Camera::Result result
...
+                    tmpColorMode = YNCNIGHT;
+                    break;
+                case Camera::ColorMode::UNPROCESSED:
+                    tmpColorMode = YNCUNPROCESSED;
+                    break;
+                default:
+                    tmpColorMode = YNCUNKNOWN;
+                    break;
+            }
+            completion(tmpColorMode, error);
```
Comment (julianoes) [inline Yuneec_SDK_iOS/YNCSDKCamera.mm]: Do we need to check if `completion` is not null?

File: Yuneec_SDK_iOS/YNCSDKCamera.mm [hunk 4]
```diff
@@ -26,6 +26,41 @@ void receive_camera_result(YNCCameraCompletion completion, Camera::Result result
...
+        }
+    }
+}
+
 #if 0
 //MARK: receive get camera setting result
```
Comment (julianoes) [inline Yuneec_SDK_iOS/YNCSDKCamera.mm]: We can delete this commented out code.

File: Yuneec_SDK_iOS/YNCSDKCamera.mm (hunk 3 above)
Comment (Sushma7785) [reply Yuneec_SDK_iOS/YNCSDKCamera.mm]: Isn't the statement if(completion), checking for null?

File: Yuneec_SDK_iOS/YNCSDKCamera.h (hunk 2 above)
Comment (Sushma7785) [reply Yuneec_SDK_iOS/YNCSDKCamera.h]: Right. YNCCameraColorModeNeutral is better.

**Comment by Sushma7785** [comment]
😄

**Pull request merged**